lib/jekyll/commands/doctor.rb

```diff
@@ -91,6 +94,15 @@ def urls_only_differ_by_case(site)
           urls_only_differ_by_case
         end
 
+        def proper_site_url?(site)
+          url = site.config["url"]
+          [
+            url_exists?(url),
+            url_valid?(url),
```

Comment:
I don't follow.

We _really_ want `site.url` to be set, either manually by the user or automatically by something like `pages-gem`. If `site.url` is blank, it is likely to cause problems and we should warn the user about this.

If a user has some special reason to use a blank `site.url` then they will understand that this warning does not apply to them.
On the other hand, if a user has simply neglected to set `site.url` and now things aren't working as expected,  this check should alert them to the issue and suggest setting `site.url`. That is the entire point of this PR.

I'm probably not understanding your concerns; can you try to explain to me again now that I've explained my intent?